detox/android/detox/src/main/java/com/wix/detox/espresso/scroll/SwipeHelper.kt

```diff
@@ -0,0 +1,114 @@
+package com.wix.detox.espresso.scroll
+
+import android.view.View
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.*
+import com.wix.detox.common.DetoxErrors
+import com.wix.detox.espresso.common.annot.*
+import com.wix.detox.espresso.utils.AgnosticPoint2D
+import kotlin.math.min
+import kotlin.math.max
+
+private const val EDGE_FUZZ_FACTOR = 0.083f
+
+typealias CreateSwipeAction = (
+        swiper: Swiper,
+        startCoordinatesProvider: CoordinatesProvider,
+        endCoordinatesProvider: CoordinatesProvider,
+        precisionDescriber: PrecisionDescriber
+) -> ViewAction
+
+class SwipeHelper(private val createAction: CreateSwipeAction) {
+
+    fun swipeInDirection(
+            direction: Int,
+            fast: Boolean = true,
+            offset: Double = Double.NaN,
+            startPositionX: Double = Double.NaN,
+            startPositionY: Double = Double.NaN
+    ): ViewAction {
+        val unsafeStartingPoint = AgnosticPoint2D.fromDoubles(startPositionX, startPositionY, direction)
+        val startingPoint = getSafeStartingPoint(unsafeStartingPoint, direction)
+        val start = translateFrom(startingPoint, direction)
+        val end = translateTo(startingPoint, direction, offset)
+        val swiper = if (fast) Swipe.FAST else Swipe.SLOW
+
+        return this.createAction(swiper, start, end, Press.FINGER)
+    }
+
+    private fun getSafeStartingPoint(unsafeStartingPoint: AgnosticPoint2D, direction: Int): AgnosticPoint2D {
+        val isDescendingMove = direction == MOTION_DIR_UP || direction == MOTION_DIR_LEFT
```

Comment:
Would you mind moving this to `MotionDefs` itself? Sounds like a common util like the existing ones (e.g. `isVertical`)